src/main/java/transforms/DateAsDateTimeTransform.java

```diff
@@ -25,10 +25,18 @@ public DateAsDateTimeTransform(
             @JsonProperty( Constants.TIMEZONE ) Optional<String> timezone
     ) {
         this.pattern = pattern;
-        if (timezone.isPresent()){
-            this.timezone = TimeZone.getTimeZone( timezone.get() );
+        if ( timezone.isPresent() ) {
+            String timezoneId = timezone.get();
```

Comment:
this code looks copy-pasted from above -- we should extract this to one central place to call instead

Reply:
done